dependency.go

```diff
@@ -45,3 +51,18 @@ func (d *Dependency) Inject() {
 	// Handler
 	d.UserHandler = api.NewUserHandler(uSvc)
 }
+
+// GracefulShutdown ... gracefully shutdown
+func (d *Dependency) GracefulShutdown(ctx context.Context, server *http.Server) {
```

Comment:
There is no need to worry about the safe connection close, as Context does everything.
So there is no need to do this with `DI`.
Example:
database/sql : https://blog.gopheracademy.com/advent-2016/database_sql/
firestore/pubsub : The Lifetime of the internal gRPC Stream is determined by Context. [google.golang.org/grpc](https://godoc.org/google.golang.org/grpc)

Reply:
ok!
I will deal with it!